test/plugins/tokens.test.js

```diff
@@ -302,6 +302,69 @@ describe('token plugin test', () => {
         });
     });
 
+    describe('PATCH /tokens/{id}/refresh', () => {
+        let options;
+
+        beforeEach(() => {
+            options = {
+                method: 'PATCH',
+                url: `/tokens/${tokenId}/refresh`,
+                credentials: {
+                    username,
+                    scope: ['user']
+                }
+            };
+        });
+
+        it('returns 404 when the token does not exist', () => {
+            tokenFactoryMock.get.resolves(null);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 404);
+            });
+        });
+
+        it('returns 404 when the user does not exist', () => {
+            userFactoryMock.get.withArgs({ username }).resolves(null);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 404);
+            });
+        });
+
+        it('returns 200 if regenerated successfully', () => {
+            const expected = Object.assign({}, testToken, { value: 'newValue' });
+
+            tokenMock.toJson.returns(Object.assign({}, tokenMock.toJson(), expected));
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 200);
+                assert.calledOnce(tokenMock.regenerate);
```

Comment:
The function in the model is still called `regenerate`

I can put a PR in models to change that